### PR TITLE
EDSC-4140: convert cypress retrievals/downloads.cy.js to playwright.

### PR DESCRIPTION
**Pull request by bnp26**
# Overview

### What is the feature?

Convert cypress retrievals/downloads.cy.js to playwright.

### What is the Solution?

Pretty straight forward. Just reproduced the cypress test in plawright.

### What areas of the application does this impact?

playwright test

# Testing

### Reproduction steps

- **Environment for testing:**localhost
- **Collection to test with:**n/a

1. run the playwright tests using ```npm run playwright:ui```
2. expand the download tab until you hit the actual test and click the green play button and run the test.
3. Run through the rest of the playwright tests and make sure nothing else breaks.

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.
image

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: tests/e2e/download/download.spec.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        })
+      } else {
+        await route.fulfill({
+          json: retrievals.body,
+          headers: retrievals.headers
+        })
+      }
+    })
+
+    await page.route(/granule_links/, async (route) => {
```
Comment (eudoroolivares2016) [inline tests/e2e/download/download.spec.js:47]:
```suggestion
       await page.route(/granule_links/, async (route) => {
      await route.fulfill({
        json: {
          cursor: 'mock-cursor',
          links: {
            browse: [],
            download: [
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1720.061.2020008170450.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1900.061.2020008170003.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/007/MYD04_3K.A2020007.1805.061.2020008182434.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/008/MYD04_3K.A2020008.1850.061.2020010183913.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/009/MYD04_3K.A2020009.1755.061.2020010200250.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/010/MYD04_3K.A2020010.1835.061.2020011153413.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/011/MYD04_3K.A2020011.1740.061.2020012150910.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/012/MYD04_3K.A2020012.1825.061.2020013152621.hdf'
            ],
            s3: []
          }
        },
        headers: authHeaders
      })
    })
```
I removed the second call to match the cypress one and the test was still passing locally

Comment (eudoroolivares2016) [reply tests/e2e/download/download.spec.js:47]: It did get to the other call though so I'm fine either way if anything I think yours is a bit more verbose and has paging more correctly mocked

**Review by eudoroolivares2016 — APPROVED** [review]

File: tests/e2e/download/download.spec.js (hunk 1 above)
Comment (bnp26) [reply tests/e2e/download/download.spec.js:47]: I think I like it the way that I wrote things.

File: tests/e2e/download/download.spec.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/007/MYD04_3K.A2020007.1805.061.2020008182434.hdf').first()).toBeVisible()
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/008/MYD04_3K.A2020008.1850.061.2020010183913.hdf').first()).toBeVisible()
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/009/MYD04_3K.A2020009.1755.061.2020010200250.hdf').first()).toBeVisible()
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/010/MYD04_3K.A2020010.1835.061.2020011153413.hdf').first()).toBeVisible()
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/011/MYD04_3K.A2020011.1740.061.2020012150910.hdf').first()).toBeVisible()
+    await expect(page.getByText('https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/012/MYD04_3K.A2020012.1825.061.2020013152621.hdf').first()).toBeVisible()
```
Comment (dpesall) [inline tests/e2e/download/download.spec.js:124]:
```suggestion
    const links = [
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1720.061.2020008170450.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1900.061.2020008170003.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/007/MYD04_3K.A2020007.1805.061.2020008182434.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/008/MYD04_3K.A2020008.1850.061.2020010183913.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/009/MYD04_3K.A2020009.1755.061.2020010200250.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/010/MYD04_3K.A2020010.1835.061.2020011153413.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/011/MYD04_3K.A2020011.1740.061.2020012150910.hdf',
      'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/012/MYD04_3K.A2020012.1825.061.2020013152621.hdf'
    ]

    await Promise.all(links.map(async (link) => {
      await expect(page.getByText(link).first()).toBeVisible()
    }))
```

**Review by dpesall — APPROVED** [review]

**Pull request merged**